### ISSUE TITLE
Add golem project hook to produce nice CDU golem template

### ISSUE THREAD

**Issue by ChrisBeeley** [CDU-data-science-team/nottshcMethods#23]
When you create a golem package you can run a project hook that can do some common set up tasks as described here https://cran.r-project.org/web/packages/golem/vignettes/f_extending_golem.html

We should have one of these for CDU golem projects, that will do the following (happy to add/ change/ debate this list):

- [x] Add the pipe to the dependencies using `usethis::use_pipe()`
- [ ] Add a `stop()` the to top of the 02.dev.R file because I source the whole thing accidentally sometimes :upside_down_face: 
- [x] Add shinydashboard as a dependency
- [ ] Add the CDU favicon using this function `golem::use_favicon("https://cdu-data-science-team.github.io/img/nottshc-cdu-data-science-logo.png")`
- [x] Host the image file above within this repo in case it moves
- [ ] Add `ext = 'png'` to `favicon()` within the `golem_add_external_resources()` function in 'app_ui.R'
- [ ] Add a shinydashboard skeleton to app_ui.R instead of vanilla Shiny UI

Being honest I don't know if all the above is even possible but I do all of the above every time I start a golem project so even half of the above would be great

**Comment by ChrisBeeley** [comment]
Logo is now helpfully hosted here https://github.com/CDU-data-science-team/assets/blob/main/logo/nottshc-cdu-data-science-logo-crop.png

**Comment by Lextuga007** [comment]
@ChrisBeeley For the package dependencies do we want all the supporting packages for the {xaringan} and {distill} to be installed?

**Comment by ChrisBeeley** [comment]
This isn't closed, is it?

**Comment by Lextuga007** [comment]
No, it's not closed - I think I might have closed it by linking to the pull request.
